recipes-asteroid/qml-asteroid-config/qml-asteroid-config_1.0.bb

```diff
@@ -0,0 +1,31 @@
+SUMMARY = "Generates a machine specific config file for AsteroidOS"
+LICENSE = "GPLv3"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/GPL-3.0;md5=c79ff39f19dfec6d293b95dea7b07891"
+
+PACKAGE_ARCH = "${MACHINE_ARCH}"
+
+python do_generate_config () {
+    machine_vars = ['MACHINE_DISPLAY_FLAT_TIRE', 'MACHINE_DISPLAY_BORDER_GESTURE_WIDTH', 'MACHINE_DISPLAY_ROUND', 'MACHINE_HAS_WLAN', 'MACHINE_HAS_SPEAKER']
+
+    from configparser import ConfigParser
+    config = ConfigParser()
+    config.optionxform=str
+    config.add_section('General')
+    for v in machine_vars:
+        if d.getVar(v) is not None:
+            config.set('General', v[len("MACHINE_"):], str(d.getVar(v)))
+
+    with open(d.getVar('WORKDIR') + "/machine.conf", "w") as machine_conf:
+        config.write(machine_conf)
+}
+
+do_install_append() {
+    if [ -f ../machine.conf ]; then
+      install -d ${D}/etc/qml-asteroid/
+      cp ../machine.conf ${D}/etc/qml-asteroid/machine.conf
```

Comment:
In principle we should use install to install all files (especially system configuration files) and explicitly set permissions.
In practice... I know it's easier to just use cp and I've done that everywhere, but I regret. :p